core/templates/dev/head/pages/exploration_player/AudioPreloaderService.js

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to preload audio into AssetsBackendApiService's cache.
+ */
+
+oppia.factory('AudioPreloaderService', [
+  '$modal', 'explorationContextService', 'AssetsBackendApiService',
+  'ExplorationPlayerStateService', 'UrlInterpolationService',
+  function($modal, explorationContextService, AssetsBackendApiService,
+      ExplorationPlayerStateService, UrlInterpolationService) {
+    // Whether all audio files have been preloaded in the exploration.
+    var _hasPreloaded = false;
+
+    // This is a file to exclude while preloading all audio translations 
```

Comment:
Just a thought: if you're supporting the "load each file individually" case, should you be maintaining a list of exclusions here instead? Maybe exposing a markFileAsAlreadyLoaded() function would be a way to do this, but if there's some way we could do something like getAlreadyLoadedFiles() from the audio caching service then that might be even better since we don't have to keep track of this in multiple places.